src/app.rs

```diff
@@ -39,6 +39,8 @@ pub struct ServiceState {
     objects: ObjectsActor,
     /// The config object.
     config: Arc<Config>,
+    /// The download service.
+    download_svc: Arc<crate::services::download::Downloader>,
```

Comment:
maybe import this type here as well

Reply:
You mean `Downloader` instead of `crate::services::download::Downloader`?  I kind of like the prefix, says a lot more about the thing, I considered `use crate::services` though.  But the crate prefix doesn't bother me that much and there's only one service so far.

But this does remind me that I think actual rust naming guidelines would suggest `services::download::Service` which is a rename I was still considering.

Reply:
Ok, this is getting quite bikesheddy now. I would actually prefer `DownloadService`, and importing that from `crate::services::DownloadService`, which itself is a re-export. Well its just personal preference ;-)

Reply:
Nothing wrong with some friendly bikeshedding.  I think https://doc.rust-lang.org/1.0.0/style/style/naming/README.html#avoid-redundant-prefixes-[rfc-356] (which we seem to explicitly call out on the rust notion page) is against `services::download::DownloadService` and pro `services::download::Service`.  Or at least that's my interpretation.  That doesn't say anything about a `services::DownloadService` re-export though.

Reply:
I’m also thinking about `type_id` and `Debug`. The first includes the complete canonical path, the latter only the struct name itself. Sure, `Debug` as the name says should not really appear in production anywhere, but its still good to disambiguate there. When you want to `use` a lot of these, or have your IDE auto-import them, it also is good to have a more unique name, or at least re-export.

Reply:
Renamed to `DownloadService`.  Didn't go with the re-export on services yet though, the indirection doesn't yet seem worth it with just one service.  Also did some more `use` of it now, though I'm not really convinced of that.